go-controller/pkg/util/ovs_unit_test.go

```diff
@@ -2169,7 +2169,7 @@ func TestAddNormalActionOFFlow(t *testing.T) {
 			}
 			mockCall.Once()
 
-			_, _, e := AddNormalActionOFFlow("somename")
+			_, _, e := AddFloodActionOFFlow("somename")
```

Comment:
So... we changed the behavior of the function, but the existing unit test still passes? What exactly is the unit test testing?

Reply:
I think its just calling the function with valid arguments and not expecting an error. It's mocking out the actualy OVS call so there is no checking of the flow or anything. Is that right @vjayaramrh ?

Reply:
@trozet Looks like all you have done is rename the function with no changes to the parameters of the function, nor have you changed the number of arguments passed to the runCmd, so the unit tests should pass as expected when given know inputs with the outputs mocked.
@danwinship we will need integration tests as well for verifying behaviors

Reply:
> Looks like all you have done is rename the function with no changes to the parameters of the function, nor have you changed the number of arguments passed to the runCmd, so the unit tests should pass as expected when given know inputs with the outputs mocked.
>  @danwinship we will need integration tests as well for verifying behaviors

Why is it acceptable for the unit test to look at the _number_ of arguments, but not for it to look at the _content_ of the arguments? Any number of different functions could pass this unit test. It doesn't validate that the function is behaving correctly in any useful way.

Reply:
Good point, It is important to look at the content of the arguments as well to the `func AddFloodActionOFFlow(bridgeName string)` as part of unit testing. However, In this particular function (https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L656-L666) , the implementation is not validating the parameters being passed though. At some point in the future, we may need to refactor the method to add those parameter validations and accordingly update the unit tests. We would definitely need integration tests/behavior tests separate from unit tests that validates behavior/interactions as well by directly making the ovs calls the tests instead of the mocks.